test/zip-dir.test.js

```diff
@@ -99,6 +100,86 @@ describe("zip-dir", function () {
     });
   });
 
+  describe("Ensures that inner folder names can be defined", function () {
+    afterEach(cleanUp);
+
+    function customCompareMethod(file, customFilePath = "", combineCustomFilePathOrReplace = true) {
+      const zipBuffer = fs.readFileSync(path.join(sampleZipPath, file));
+      const combinedFilePath = path.join(customFilePath, file);
+      // Compare the different file buffers.
+      const fileBuffer = fs.readFileSync(
+        path.join(
+          outputPath,
+          combineCustomFilePathOrReplace ? combinedFilePath : customFilePath
+        )
+      );
+
+      expect(bufferEqual(zipBuffer, fileBuffer)).to.be.ok;
+    }
+
+    //Had to do the below, since otherwise the file did not zip in enough time.
+    function customZippingFunction(options, customFilePath) {
+      return new Promise((resolve, reject) => {
+        zipDir(
+          customFilePath || sampleZipPath,
+          options,
+          async function (err, buffer) {
+            if (err) {
+              reject(err);
+              return;
+            }
+            await fs
+              .createReadStream(xpiPath)
+              .pipe(unzip.Extract({ path: outputPath }))
+              .on("entry", (entry) => entry.autodrain())
+              .promise()
+              .then(() => resolve())
+              .catch(e=>reject(e));
+          }
+        );
+      });
+    }
```

Comment:
Had to wrap the zipping command into its own promise since it was being flaky when running tests, since the zipping often took way too long, and zipDir cannot directly be promised.